src/pages/Search/SearchPage.tsx

```diff
@@ -818,6 +828,15 @@ function SearchPage({route}: SearchPageProps) {
         }
     }, []);
 
+    useEffect(() => {
+        if (!queryJSON || !isDataMissing || !isFocused) {
+            return;
+        }
+        handleSearchAction({queryJSON, searchKey, offset: 0, shouldCalculateTotals});
+        // eslint-disable-next-line react-compiler/react-compiler
```

Comment:
I know it's not yet enforced anywhere, but we really shouldn't integrate any new code that disables the compiler, especially so high up in the components' tree. Can you rethink how this can be implemented following the rules of React?

Reply:
I agree that we should follow that approach but in this particular case I don’t think it will make much difference because the SearchPage already has a lot of issues with the React compiler even without this useEffect, which means it isn’t being compiled anyway. So adding or removing this useEffect won’t really change that situation.